cfgov/v1/forms.py

```diff
@@ -1,12 +1,86 @@
+import time
+from datetime import timedelta
+
 from django import forms
 from django.forms.utils import ErrorList
 from django.forms.widgets import DateInput, SelectMultiple
+from django.contrib.auth import get_user_model
+from django.utils import timezone
+from django.db.models import Count
+from django.core.exceptions import ValidationError
+from django.contrib.auth import authenticate, get_user_model
+from django.conf import settings
 
 from sheerlike.query import QueryFinder
 from sheerlike.templates import date_formatter
 from v1.models.events import EventPage
 from util import ERROR_MESSAGES
 
+from .models import base
+from .models.base import PasswordHistoryItem
+
+# importing wagtail.wagtailadmin.forms at module load time seems to have
+# caused some trouble with the translation subsystem being invoked before
+# it's ready-- so I've wrapped it in a function
+def login_form():
+    from wagtail.wagtailadmin import forms as wagtail_adminforms
+    from v1.util import password_policy
+
+    class LoginForm(wagtail_adminforms.LoginForm):
+
+	def clean(self):
+	    username = self.cleaned_data.get('username')
+	    password = self.cleaned_data.get('password')
+
+	    if username and password:
+		self.user_cache = authenticate(username=username,
+					       password=password)
+		if (self.user_cache is None and username is not None):
+		    UserModel = get_user_model()
+		    user = UserModel._default_manager.get(username=username)
```

Comment:
This needs to be protected for when there's not user for the username given. Getting this: `User matching query does not exist.`